pkg/controller/generic_reconciler.go

```diff
@@ -257,16 +283,21 @@ func (gr *GenericReconciler) lookUpType(obj *unstructured.Unstructured) (runtime
 	return typedObj, nil
 }
 
-func (gr *GenericReconciler) handleResourceDeletions() {
-	for k := range *gr.objectValidationCache {
-		if gr.currentObjects.has(k) {
+func (gr *GenericReconciler) handleResourceDeletions(ctx context.Context, gvk schema.GroupVersionKind) {
+	currentObjects := ctx.Value(ctxKeyCurrentObjects).(*validationCache)
+	objectHistory, ok := (*gr.objectValidationCache)[cacheSetKey(gvk)]
+	if !ok {
+		return
+	}
+	for k := range *objectHistory {
+		if currentObjects.has(k) {
 			continue
 		}
-		validations.DeleteMetrics(k.namespace, k.name, k.kind)
-		gr.objectValidationCache.removeKey(k)
-
+		nsUID := gr.watchNamespaces.getNamespaceUID(k.namespace)
+		validations.DeleteMetrics(k.namespace, nsUID, k.name, k.uid, gvk.Kind)
+		objectHistory.removeKey(k)
 	}
-	gr.currentObjects.drain()
+	currentObjects = nil
```

Comment:
What is this supposed to do?
This just set's the local scoped variable to `nil`

Reply:
oops